source/agora/consensus/EnrollmentManager.d

```diff
@@ -532,6 +532,92 @@ public class EnrollmentManager
         return false;
     }
 
+    /***************************************************************************
+
+        Get all the valid validators
```

Comment:
It means the validator that has valid `enrolled_height` value not null, and has a validator cycle not yet expired. I could find more obvious sentence.

Reply:
>  and has its validator cycle not yet expired

The code and the documentation disagree then. Because the code only checks that `enrolled_height` is not null.

Reply:
You're right but it is already checked whether the validator cycle has been expired in the `setCurrentBlockHeight` function. I have just changed the sentence to 'Get all the current validators'.

Reply:
When can the `enroled_height` have a null value?

Reply:
I see. It's set when `updateEnrolledHeight()` is running.